### PR TITLE
Refactor flutter_plugins

### PR DESCRIPTION
**Pull request by sealesj**
Refactor flutter_plugins suite in order to reduce testing logic in test.dart and allow for later implementing package:test onto the existing flutter_plugins tests

Also update the test_test invocation of getFlutterPackagesVersion

Part of https://github.com/flutter/flutter/issues/145482

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

File: dev/bots/suite_runners/run_flutter_packages_tests.dart [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import 'dart:io' show Directory, File;
+
+import 'package:file/file.dart' as fs;
+import 'package:file/local.dart';
+import 'package:path/path.dart' as path;
+
+import '../run_command.dart';
+import '../test.dart';
+import '../utils.dart';
+
```
Comment (christopherfujino) [inline dev/bots/suite_runners/run_flutter_packages_tests.dart:14]: nit delete extra line

File: dev/bots/suite_runners/run_flutter_packages_tests.dart [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+/// Executes the test suite for the flutter/packages repo.
+Future<void> flutterPackagesRunner(String flutterRoot) async {
+
+  Future<void> runAnalyze() async {
+    printProgress('${green}Running analysis for flutter/packages$reset');
+    final Directory checkout = Directory.systemTemp.createTempSync('flutter_packages.');
+    await runCommand(
+      'git',
+      <String>[
```
Comment (christopherfujino) [inline dev/bots/suite_runners/run_flutter_packages_tests.dart:24]: nit const

File: dev/bots/suite_runners/run_flutter_packages_tests.dart [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    );
+    // Prep the repository tooling.
+    // This test does not use tool_runner.sh because in this context the test
+    // should always run on the entire packages repo, while tool_runner.sh
+    // is designed for flutter/packages CI and only analyzes changed repository
+    // files when run for anything but master.
+    final String toolDir = path.join(checkout.path, 'script', 'tool');
+    await runCommand(
+      'dart',
+      <String>[
```
Comment (christopherfujino) [inline dev/bots/suite_runners/run_flutter_packages_tests.dart:52]: nit const

**Review by goderbauer — APPROVED** [review]
LGTM

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**